docs/reference/api/usb/usb_component.md

```diff
@@ -0,0 +1,89 @@
+<h2 id="usb-overview">USB components overview</h2>
+
+The Universal Serial Bus (USB) is the most widely used bus in today's computer. USB has been designed to standardize connections between the computer and peripherals. For example, keyboards, mice, USB audio devices, printers, scanners, disk drives and cameras can use the same bus to exchange data with a computer.
+
+This document refers to the Mbed OS classes providing USB peripheral functionality as USB components. These USB components inherit from USBDevice and provide specific USB peripherial functionality. Below is information common to all of the Mbed OS USB components and how to use them.
+
+### Component construction
+
+Constructing a USB component with the default parameters perform initialization for you. Once the constructor is finished then the device is ready to be used. This process:
+
+- Selects the built-in USBPhy for your hardware.
+- Starts the USB connection sequence.
+- Blocks until everything is ready.
+
+An example of USBKeyboard's default constructor:
+
+```
+#include "mbed.h"
+#include "USBKeyboard.h"
+
+// Default constructor to connect and block until ready
+USBKeyboard key;
+
+int main(void)
+{
+    while (1) {
+        key.printf("Hello World\r\n");
+        wait(1);
+    }
+}
+```
+
+The first optional parameter to the constructor of all USB components is a bool that specifies whether the USB component should connect and block until ready. By default, this value is `true`. When set to `false`, USB still selects the built-in USBPhy for your hardware, but it does not block or start the connection sequence. This is useful when you do not want to connect USB at boot.
```

Comment:
Query: What does "should connect" mean? Must connect? Why should it? Or why shouldn't it?